File: DSA/c/linkedlist.c
#include<stdio.h>
#include<stdlib.h>
typedef struct Node{
  int value;
  struct Node *next;
} Node;
typedef struct {
  Node *head;
} LinkedList;
void append(LinkedList *list, int i);
int main(){
  LinkedList l1 ;
  l1.head = NULL;
  append(&l1, 20);
  append(&l1, 10);
  printf("%d %d\n",
         l1.head->value,
         l1.head->next->value
  );
  return 0;
}
void append(LinkedList *list, int i){
  Node *n = malloc(sizeof(Node));
  if(!n) exit(1);
  n->value = i;
  n->next = NULL;
  if(list->head == NULL){
    list->head = n;
  } else {
    Node *curr = list->head;
    while(curr->next){
      curr = curr->next;
    }
    curr->next = n;
  }
}

File: DSA/python/linkedlist.py
class Node:
    def __init__(self, value, next=None):
        self.value = value
        self.next = next


class LinkedList:
    def __init__(self, headValue=None):
        self.head = Node(headValue)
        self.tail = self.head

    def __str__(self):
        s = "LinkedList ("
        next = self.head
        while next is not None:
            if type(next.value) is str:
                s += f'"{next.value}"'
            else:
                s += str(next.value)
            if next.next is not None:
                s += " -> "
            next = next.next
        s += ")"
        return s

    def append(self, value, next=None):
        self.tail.next = Node(value, next)
        self.tail = self.tail.next

    def prepend(self, value, next=None):
        self.head = Node(value, self.head)

    def pop(self):
        curr = self.head
        while curr.next.next is not None:
            curr = curr.next
        curr.next = None

    def pop_head(self):
        if self.head:
            self.head = self.head.next

    def clear(self):
        self.head = None


if __name__ == "__main__":
    a = LinkedList(10)
    a.append(20)
    a.append(40)
    a.prepend("Houli")
    print("After prepend: ", a)
    a.pop()
    print("After pop:", a)
    a.pop_head()
    print("Pop head:", a)
    a.clear()
    print("After clear:", a)
